.github/pull_request_template.md

```diff
@@ -0,0 +1,42 @@
+### Description
+
+Please include a summary of the change being made.
+
+Fixes #(issue)
+
+### Why is the change necessary?
+
+What capability does it enable? What problem does it solve?
+
+### Solution
+
+Please include an overview of the solution. Discuss trade-offs made, caveats, alternatives, etc.
+
+### Testing
+
+How was this change tested?
+
+----
+
+### Pull Request Checklist
+
+Please check all boxes (including N/A items)
+
+#### Testing
+
+- [ ] Unit tests added
+- [ ] integration test added
+
+#### Documentation
+
+- [ ] __docs__: All relevant [docs](https://github.com/aws/aws-step-functions-data-science-sdk-python/tree/main/doc) updated
+- [ ] __docstrings__: All public APIs documented
+
+### Title and description
+
+- [ ] __Change type__: Title is prefixed with change type: and follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
```

Comment:
I don't really care for conventional commits because they use up valuable space in an already short commit message. The more useful thing is a well-written commit message. https://chris.beams.io/posts/git-commit/

Not strongly opinionated here though. You and @ca-nguyen have been using conventional commits consistently and I am ok continuing that pattern.
 
Our contributing guide does [mention prefixes](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/CONTRIBUTING.md#committing-your-change), but not `chore` and `feat`

Reply:
Absolutely. A well-written commit message is absolutely necessary and these 2 things are not mutually exclusive.

My intention behind asking for conventional commits, is more to automate changelog generation. There is a ton of [tooling](https://www.conventionalcommits.org/en/about/#tooling-for-conventional-commits) that helps automate it and that's my motivation. I do like it more than you, but it's not nearly as essential as a well written commit message :)